components/ILIAS/Calendar/classes/class.ilDatePresentation.php

```diff
@@ -85,13 +99,15 @@ public static function resetToDefaults()
         self::setUseRelativeDates(true);
     }
 
-    public static function formatDate(ilDateTime $date, bool $a_skip_day = false, bool $a_include_wd = false, bool $include_seconds = false): string
-    {
+    public static function formatDate(
+        ilDateTime $date,
+        bool $a_skip_day = false,
+        bool $a_include_wd = false,
+        bool $include_seconds = false,
```

Comment:
Why not adding an optional language user object and deriving the appropriate language object from that?

Reply:
The static Method setLanguage is already in use in many components. Some of which have not a clear direct formatDate() call. Without knowing the side effects, it's difficult to now change that.
I agree that passing this is a much better practice, but as it is already used, I cannot predict what effects that may have if we remove setLanguage(). And having both setLanguage and a parameter seems to open us up to race condition and confusion for developers.

Let me know your thoughts to this. Also, maybe start a discussion with other maintainers if that seems like a greater deal.

Best @fhelfer